clients/da-vinci-client/src/main/java/com/linkedin/davinci/kafka/consumer/StoreAwarePartitionWiseKafkaConsumerService.java

```diff
@@ -0,0 +1,200 @@
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+import org.apache.logging.log4j.LogManager;
+
+
+/**
+ * {@link StoreAwarePartitionWiseKafkaConsumerService} is used to allocate share consumer from consumer pool at partition
+ * granularity. One shared consumer may have multiple topics, and each topic may have multiple consumers.
+ * This is store-aware version of topic-wise shared consumer service. The topic partition assignment in this service has
+ * a heuristic that we should distribute the all the subscriptions related to a same store / version as even as possible.
+ * The load calculation for each consumer will be:
+ * Hybrid store: Consumer assignment size + STORE_SUBSCRIPTION_LOAD * subscription count for the same store;
+ * Batch only store: Consumer assignment size + STORE_SUBSCRIPTION_LOAD * subscription count for the same store version;
+ * and we will pick the least loaded consumer for a new topic partition request.
+ */
+public class StoreAwarePartitionWiseKafkaConsumerService extends PartitionWiseKafkaConsumerService {
+  private static final long DEFAULT_STORE_SUBSCRIPTION_LOAD = 100;
+  private final Map<String, Map<PubSubConsumerAdapter, Integer>> resourceIdentifierToConsumerMap =
+      new VeniceConcurrentHashMap<>();
+  private final ReadOnlyStoreRepository storeRepository;
+
+  StoreAwarePartitionWiseKafkaConsumerService(
+      final ConsumerPoolType poolType,
+      final PubSubConsumerAdapterFactory consumerFactory,
+      final Properties consumerProperties,
+      final long readCycleDelayMs,
+      final int numOfConsumersPerKafkaCluster,
+      final IngestionThrottler ingestionThrottler,
+      final KafkaClusterBasedRecordThrottler kafkaClusterBasedRecordThrottler,
+      final MetricsRepository metricsRepository,
+      final String kafkaClusterAlias,
+      final long sharedConsumerNonExistingTopicCleanupDelayMS,
+      final TopicExistenceChecker topicExistenceChecker,
+      final boolean liveConfigBasedKafkaThrottlingEnabled,
+      final PubSubMessageDeserializer pubSubDeserializer,
+      final Time time,
+      final AggKafkaConsumerServiceStats stats,
+      final boolean isKafkaConsumerOffsetCollectionEnabled,
+      final ReadOnlyStoreRepository metadataRepository,
+      final boolean isUnregisterMetricForDeletedStoreEnabled) {
+    super(
+        poolType,
+        consumerFactory,
+        consumerProperties,
+        readCycleDelayMs,
+        numOfConsumersPerKafkaCluster,
+        ingestionThrottler,
+        kafkaClusterBasedRecordThrottler,
+        metricsRepository,
+        kafkaClusterAlias,
+        sharedConsumerNonExistingTopicCleanupDelayMS,
+        topicExistenceChecker,
+        liveConfigBasedKafkaThrottlingEnabled,
+        pubSubDeserializer,
+        time,
+        stats,
+        isKafkaConsumerOffsetCollectionEnabled,
+        metadataRepository,
+        isUnregisterMetricForDeletedStoreEnabled);
+    this.LOGGER =
```

Comment:
This is not a good pattern as the constructor of the super class will construct a `LOGGER` and get overrided right away here.
We can pass a `Logger name` into the constructor, so that only the super class will be in charge of logger initialization, no here.